extensions/api/blocking_api.py

```diff
@@ -30,23 +33,38 @@ def get_model_info():
 
 
 class Handler(BaseHTTPRequestHandler):
+    def authentication(self):
+        if not shared.args.http_auth:
+            return True
+
+        AUTHENTICATED = False
+        if "X-TEXT-UI-AUTH" in self.headers:
+            authz = self.headers['X-TEXT-UI-AUTH']
+            if str(authz) == str(AUTH_TOKEN):
+                AUTHENTICATED = True
+        return AUTHENTICATED
+
     def do_GET(self):
-        if self.path == '/api/v1/model':
+        is_auth = self.authentication()
```

Comment:
The self.send_error function does return control flow back and the other code is executed, hence the repeated checks, as otherwise users would not see the results but their actions would impact the system.

Reply:
Make sense. What if you check it first `if` of the `if/elif/else` block? Wouldn't it return directly after failing?

I have in mind that people will forget to add the `and is_auth`.

Reply:
The change proposed by @Vincent-Stragier makes sense to me. It would be better to have a single check than to add the same check multiple times.